### PR TITLE
Bump to pydantic 2.8.0

### LINKED ISSUE #701: Upgrade pydantic dependency to v2
**Is your feature request related to a problem? Please describe.**

Current version of healthchecks.io is incompatible with pydantic 2. There is an auto-update from dependabot in https://github.com/andrewthetechie/py-healthchecks.io/pull/700 however, due to major version upgrade, we need to do a bit more work to make it compatible with pydanticv2.

I'm using py-healthchecks.io along with another library that depends on pydantic v2, and due to the version conflict, I'm currently unable to use py-healthchecks.io.

**Describe the solution you'd like**
It would be great for py-healthchecks.io to support pydantic v2

**Describe alternatives you've considered**
We could keep it as is, but I imagine more and more libraries will be upgrading their pydantic dependency. Therefore, in order to keep py-healthchecks.io usable for other users, I recommend upgrading the version.

### PR DESCRIPTION
**Pull request by Mariatta**
This PR includes the dependabot change from https://github.com/andrewthetechie/py-healthchecks.io/pull/700

- Ran the `bump-pydantic` tool.
- Add a few safety ignores: jinja2, and requests
- Bump anyio dependency
- Include packaging dependency. It seems to be required.

Closes #701

### CODE REVIEW

File: src/healthchecks_io/client/sync_client.py [hunk 1]
```diff
@@ -302,6 +302,7 @@ def get_badges(self) -> Dict[str, badges.Badges]:
 
         svg, json, shields: reports two states: "up" and "down". It considers any checks in the grace period
             as still "up".
+
```
Comment (Mariatta) [inline src/healthchecks_io/client/sync_client.py:305]: This fixes the warning message from docs-build.

File: src/healthchecks_io/schemas/checks.py [hunk 2]
```diff
@@ -14,34 +14,36 @@
...
+    update_url: Optional[str] = None
+    pause_url: Optional[str] = None
+    channels: Optional[str] = None
+    timeout: Optional[int] = None
+    uuid: Optional[str] = None
+
+    # TODO[pydantic]: We couldn't refactor the `validator`, please replace it by `field_validator` manually.
+    # Check https://docs.pydantic.dev/dev-v2/migration/#changes-to-validators for more information.
     @validator("uuid", always=True)
     def validate_uuid(cls, value: Optional[str], values: Dict[str, Any]) -> Optional[str]:  # noqa: B902
```
Comment (andrewthetechie) [inline src/healthchecks_io/schemas/checks.py:48]: Validator here that needs to be converted to a field_validator

**Review by andrewthetechie — CHANGES_REQUESTED** [review]
One last validator that needs conversion to a field_validator.

Thanks for submitting the PR!

**Comment by Mariatta** [comment]
Thanks @andrewthetechie. I've updated the `validate_uuid` using `field_validator` decorator. Please check again.

**Review by andrewthetechie — APPROVED** [review]

File: src/healthchecks_io/schemas/checks.py (hunk 2 above)
Comment (Mariatta) [reply src/healthchecks_io/schemas/checks.py]: Good catch! I've updated this and removed the TODO.

**Pull request merged**